tests/unit/errors/test_common_functionality.py

```diff
@@ -117,23 +117,44 @@ def test_info_is_falsey_on_non_dict_json(make_json_response):
     assert str(err.info) == "AuthorizationParameterInfo(:)|ConsentRequiredInfo(:)"
 
 
-def test_consent_required_info(make_json_response):
-    res = make_json_response(
-        {"code": "ConsentRequired", "required_scopes": ["foo", "bar"]}, 401
-    )
+@pytest.mark.parametrize(
+    "body, is_detected, required_scopes",
+    (
+        (
+            {"code": "ConsentRequired", "required_scopes": ["foo", "bar"]},
+            True,
+            ["foo", "bar"],
+        ),
+        (
+            {"code": "ConsentRequired", "required_scope": "foo bar"},
+            True,
+            ["foo bar"],
+        ),
+        ({"code": "ConsentRequired"}, False, None),
+        ({"code": "ConsentRequired", "required_scopes": []}, False, None),
+        ({"code": "ConsentRequired", "required_scopes": ["foo", 123]}, False, None),
+        ({"code": "ConsentRequired", "required_scope": 1}, False, None),
+    ),
+)
+def test_consent_required_info(make_json_response, body, is_detected, required_scopes):
+    res = make_json_response(body, 401)
```

Comment:
I'm not sure that this matters as a practical concern here, but out of curiosity, why is this 401 rather than 403? It seems like the docs suggest this should canonically be a 403.

Reply:
Yeah, it should be a 403 if we want to match what services do, but it doesn't matter much since the ConsentRequiredInfo detection isn't keyed to any particular status code.

I'll switch it since we're here though. :+1: